src/Form/KeyForm.php

```diff
@@ -111,6 +111,23 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
   /**
    * {@inheritdoc}
    */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    // Only run key settings validation if the form is being submitted
+    if($form_state->isSubmitted()) {
+      $plugin_settings = (new FormState())->setValues($form_state->getValue('key_settings'));
```

Comment:
Couldn't this just be `$plugin_settings = $form_state->getValue('key_settings');`?